intezer.py

```diff
@@ -971,9 +975,24 @@ def _handle_subanalyses(
 
                 if self.config.get("download_subfiles", True):
                     if can_we_download_files or self.config.get("try_to_download_every_file", False):
-                        file_was_downloaded = self.client.download_file_by_sha256(sub_sha256, self.working_directory)
+
+                        file_was_downloaded = False
+                        path = f"{os.path.join(self.working_directory, sub_sha256)}.sample"
+
+                        if self.privileged == "true":
+                            # Attempt to download from AL4 filestore first.
+                            # This prevents un-necessary hit against user's quota with Intezer
+                            fs = forge.get_filestore()
```

Comment:
I think you should create the filestore object in the `__init__` so that it does not get recreated for each subfile and for each file analyzed by the service.

Reply:
PR updated as requested. I also added a couple of debug logs to validate more easily.